geode-core/src/main/java/org/apache/geode/pdx/internal/json/PdxToJSON.java

```diff
@@ -226,7 +225,8 @@ private void getJSONStringFromArray(JsonGenerator jg, Object value, String pf)
       jg.writeEndArray();
     } else {
       throw new IllegalStateException(
-          "PdxInstance returns unknown pdxfield " + pf + " for type " + value);
+          "PdxInstance returns unsupported type " + value.getClass()
```

Comment:
The caller already made sure value.getClass().isArray() was true. So this message should say something about the array element type not being supported by JSON. Given a Class instance that isArray() you can call getComponentType() it to get the class that describes the element type. The message could say that the componentType is not supported for JSON. Also keep in mind that "value" is an array and toString on an array is not very helpful (just prints out the class name and its hashCode but not the elements of the array).
Also you need a unit test for this one

Reply:
Does this work?
```
          "The pdx field " + pf + " is an array whose component type "
              + value.getClass().getComponentType()
              + " can not be converted to JSON.");
```
I'm not sure I like trusting that the caller made sure that value.getClass().isArray() was true though.

Reply:
*I* like this better:
```
    } else if (value.getClass().isArray()) {
      throw new IllegalStateException(
          "The pdx field " + pf + " is an array whose component type "
              + value.getClass().getComponentType()
              + " can not be converted to JSON.");
    } else {
      throw new IllegalStateException(
          "Expected an array for pdx field " + pf + ", but got an object of type "
              + value.getClass());
    }
```

Reply:
looks good

Reply:
Unit test added